pkg/backend/aws/ecs.go

```diff
@@ -42,20 +43,54 @@ func (b *Backend) DescribeService(ctx context.Context, serviceName *string) (*ec
 	return &out.Services[0], nil
 }
 
-func (b *Backend) GetServiceTasks(ctx context.Context, serviceName *string) ([]string, error) {
+// GetECSTasksForStackService returns the task ARNs associated with a particular happy stack and service.
+// The filter is based on the name of the stack and the service name provided in the docker-compose file.
+// For now, the tasks ARNs are return for services that have a name with the stack and
+// docker-compose service name in them.
+func (b *Backend) GetECSTasksForStackService(ctx context.Context, serviceName, stackName string) ([]string, error) {
 	clusterARN := b.integrationSecret.ClusterArn
-	out, err := b.ecsclient.ListTasks(ctx, &ecs.ListTasksInput{
-		Cluster:     &clusterARN,
-		ServiceName: serviceName,
+	ls, err := b.ecsclient.ListServices(ctx, &ecs.ListServicesInput{
+		Cluster: &clusterARN,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list ECS services for a stack")
+	}
+
+	ds, err := b.ecsclient.DescribeServices(ctx, &ecs.DescribeServicesInput{
+		Cluster:  &clusterARN,
+		Services: ls.ServiceArns,
 	})
 	if err != nil {
-		var snfe *ecstypes.ServiceNotFoundException
-		if errors.As(err, &snfe) {
-			return []string{}, nil
+		return nil, errors.Wrap(err, "unable to describe ECS services for stack")
+	}
+
+	// TODO: right now, happy has no control over what these ECS services are called
+	// but a convention has started where the stack name is a part of the service name
+	// and so is the docker-compose service name. Usually, its of the form <stackname>-<docker-compose-service-name>.
+	// I'm being a little loose with this matching to account for this convention and also
+	// give us some wiggle room to change this convention in the future.
+	stackServNames := []string{}
+	for _, s := range ds.Services {
+		if strings.Contains(*s.ServiceName, serviceName) && strings.Contains(*s.ServiceName, stackName) {
```

Comment:
serviceName passed into this method is a concatenation of a service name and a stack name, can this be simplified?